app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -58,6 +65,24 @@ def finalize(*args)
         check_exit_status
       end
 
+      def self.feature_job_event_name(label, suffix = :success)
+        ::Foreman::Observable.event_name_for("#{::Actions::RemoteExecution::RunHostJob.event_name_base}_#{label}_#{::Actions::RemoteExecution::RunHostJob.event_name_suffix(suffix)}")
+      end
+
+      def emit_feature_event(execution_plan, hook = :success)
+        return unless root_action?
+
+        if input["job_features"]
+          input['job_features'].each do |feature|
+            name = "#{self.class.event_name_base}_#{feature}_#{self.class.event_name_suffix(hook)}"
+            trigger_hook name, payload: event_payload(execution_plan)
+          end
+        else
+          name = "#{self.class.event_name_base}_#{self.class.event_name_suffix(hook)}"
+          trigger_hook name, payload: event_payload(execution_plan)
```

Comment:
Can the else branch ever happen? And if it does and if I'm reading it right, then it would emit the same generic event that is already being emitted since `RunHostJob` is an observable action. In any case, I'd vote for dropping it so it either emits events for features or nothing at all

Reply:
Oh yeah I missed your point, I get it now sorry.